Toggl.Foundation.MvvmCross/ViewModels/Settings/SendFeedbackViewModel.cs

```
@@ -0,0 +1,100 @@
+
+        // Outputs
+        public IObservable<bool> IsFeedbackEmpty { get; }
+        public IObservable<bool> ErrorViewVisible { get; }
+        public IObservable<bool> SendEnabled { get; }
+        public IObservable<bool> IsLoading { get; }
+
+        // Actions
+        public UIAction CloseFeedbackAction { get; }
+
+        public SendFeedbackViewModel(
+            IMvxNavigationService navigationService,
+            IInteractorFactory interactorFactory,
+            IDialogService dialogService
+        )
+        {
+            Ensure.Argument.IsNotNull(interactorFactory, nameof(interactorFactory));
+            Ensure.Argument.IsNotNull(navigationService, nameof(navigationService));
+            Ensure.Argument.IsNotNull(dialogService, nameof(dialogService));
+
+            IsFeedbackEmpty = FeedbackText
+                .Select(text => string.IsNullOrEmpty(text))
+                .DistinctUntilChanged()
+                .StartWith(true);
+
+            SendEnabled = Observable.CombineLatest(
+                IsFeedbackEmpty,
+                isLoading.AsObservable(),
+                (isEmpty, isLoading) => !isEmpty && !isLoading
+            );
+
+            CloseFeedbackAction = new UIAction(() =>
+            {
+                return IsFeedbackEmpty.SelectMany(isEmpty => isEmpty ?
+                        Observable.Return(true) : dialogService.ConfirmDestructiveAction(ActionType.DiscardFeedback))
+                    .Where(shouldDiscard => shouldDiscard)
+                    .Do(_ => { navigationService.Close(this, false); })
+                    .SelectUnit();
+            });
+
+            ErrorViewTapped
+                .Do(e => errorViewVisible.OnNext(false))
+                .Subscribe()
+                .DisposedBy(disposeBag);
+
+            SendButtonTapped
+                .WithLatestFrom(FeedbackText, (_, text) => text)
+                .Do(_ => { isLoading.OnNext(true); })
```

Shouldn't we also hide the error at this point (if one is shown)?